### ISSUE TITLE
Automatically hide terminal panel when a file in the explorer is clicked

### ISSUE THREAD

**Issue by driesvints** [microsoft/vscode#131319]
At the moment I'm using my panels maximized always with `"workbench.panel.opensMaximized": "always",` in my settings because I'd like to use my Terminal at full height all the time.

I'm using the following keyboard shortcut to toggle the terminal window:

```
    {
        "key": "shift+cmd+w",
        "command": "workbench.action.terminal.toggleTerminal"
    },
```

The problem now exists that if I click a file in the explorer, the terminal doesn't go away but actually resizes only a bit or half-screen (depends from time to time). See the behavior in the video below:

https://user-images.githubusercontent.com/594614/130325753-78d4e905-7b03-44bb-887a-5ed6fd66938e.mov

Ideally I wish that the terminal would close when I click on a file in the explorer since the action to open up a file should bring that file up full screen. 

To summarize: I never want to use my terminal half-screen.

**Comment by bamurtaugh** [comment]
cc @sbatten this is the one we discussed last week

**Comment by khannaudit** [comment]
Been waiting for this for a year :(

**Comment by leep-frog** [comment]
Just stumbled across this issue myself.  Wrote a [simple extension](https://marketplace.visualstudio.com/items?itemName=groogle.termin-all-or-nothing) that tries to solve this.  It takes care of the main use cases (via file explorer and quick open navigation), but was written in haste so may still have a few bugs to work out.

**Comment by drewgingerich** [comment]
I would love to see a setting that minimizes the panel when opening a new file. I love the workflow of toggling a maximized terminal via hotkey, but feel friction when opening files because the new file opens behind the terminal and also the terminal becomes unmaximized. This is especially frustrating when I open a file by clicking on a link in the terminal, e.g. to jump to a problematic line from the output of a failing unit test.

**Comment by shellwhale** [comment]
I use [vscode-harpoon](https://github.com/tobias-z/vscode-harpoon) and was facing the same issue. Using @leep-frog extension works great with it.

**Comment by hassan-nmh-2244** [comment]
+1
Need this setting inside vscode please

**Comment by rowanoulton** [comment]
> Just stumbled across this issue myself. I wrote a [simple extension](https://marketplace.visualstudio.com/items?itemName=groogle.termin-all-or-nothing) that tries to solve this. It takes care of the main use cases (via file explorer and quick open navigation), but was written in haste so may still have a few bugs to work out.
> 
> > [11/14/23 Update]: I've been able to put more time into this extension over the past few months and it works reliably for the main terminal actions that typically cause a split view:
> > 
> > * Open file from File Explorer
> > * Open file from `code filename.txt`
> > * Open file from Command Prompt
> > * Ctrl+click on file reference

Would love to see built-in support for this, but in the meantime @leep-frog you're my hero, thank you!

**Comment by pookdeveloper** [comment]
+1 And the same when selecting a file when searching for a text string.
